### PR TITLE
[Kraken] Revert 2818 and split sn proximity list

### PR DESCRIPTION
**Pull request by xlqian**
This pr is a fusion of https://github.com/CanalTP/navitia/pull/2825 and https://github.com/CanalTP/navitia/pull/2813

### CODE REVIEW

File: source/type/data.cpp
Comment (pbougue) [inline source/type/data.cpp]: :warning: If we don't merge it in the same navitia version than https://github.com/CanalTP/navitia/pull/2827/files#diff-930d109fe1ca9c60e82996bda4dd2f8aR66 we will have to increment again
(the #2827 should be merged as soon as merging is open again)

File: source/georef/georef.h
Comment (pbougue) [inline source/georef/georef.h]: minor typo
```suggestion
    // We create one proximity list for each mode
```

File: source/proximity_list/proximity_list.h
Comment (pbougue) [inline source/proximity_list/proximity_list.h]: clang warning : as this is now virtual, please move the definition in .h to avoid vtable emission.

File: source/georef/tests/builder.h
Comment (pbougue) [inline source/georef/tests/builder.h]: clang warning : extra `;`

File: source/proximity_list/proximity_list.cpp
Comment (pbougue) [inline source/proximity_list/proximity_list.cpp]: clang warning: Please choose between a struct (.h) and a class (.cpp) :wink:

File: source/proximity_list/proximity_list.cpp
Comment (pbougue) [inline source/proximity_list/proximity_list.cpp]: clang warning:
```suggestion
                                          pow(radius * double(factor), 2), search_param);
```

File: source/proximity_list/proximity_list.cpp
Comment (pbougue) [inline source/proximity_list/proximity_list.cpp]: clang warning:
```suggestion
                     * asin(radius / (2. * GeographicalCoord::EARTH_RADIUS_IN_METERS)) / radius;
```

File: source/proximity_list/proximity_list.cpp
Comment (pbougue) [inline source/proximity_list/proximity_list.cpp]: clang warning: Please choose between a struct (.h) and a class (.cpp) :wink:

**Review by pbougue — COMMENTED** [review]
Mostly clang warning.
This is fat, I would avoid merging it in a big release, but...

I did not review it all, as it's too big and hard to read by commit.
I don't know if it's possible to improve that : on that PR and on future ones. As we are not on the same team, it's hard to keep track of work.

**Comment by AurelienLP** [comment]
@pbougue, what version of clang do you use ? Do you add some warning flags for compilation ?
Other than that, everything is ok for me since I have read every commit of https://github.com/CanalTP/navitia/pull/2813. I have even checked the whole PR to see if everything was alright again.

> @pbougue's reply :
> I'm on debian 9, using `clang version 3.8.1-24` from system
> Using flags defined from CMake : https://github.com/CanalTP/navitia/blob/a17048519b9a20908be26c6f0c710b3ec00984ea/source/cmake_modules/CompilationFlags.cmake#L42-L50

**Review by AurelienLP — APPROVED** [review]

**Comment by xlqian** [comment]
Jenkins, test please!

**Comment by kinnou02** [comment]
Jenkins, test this please.

File: source/georef/georef.h
Comment (kinnou02) [inline source/georef/georef.h]: we want to keep `visible` :)

File: source/kraken/data_manager.h
Comment (kinnou02) [inline source/kraken/data_manager.h:141]: You couldn't serialize it?

File: source/kraken/maintenance_worker.cpp
Comment (kinnou02) [inline source/kraken/maintenance_worker.cpp:205]: Minor: not the scope of this PR but we should move that in `data_manager`

**Review by kinnou02 — CHANGES_REQUESTED** [review]
Merge issue with https://github.com/CanalTP/navitia/pull/2827
You should probably rebase

File: source/georef/georef.h
Comment (xlqian) [reply source/georef/georef.h]: my bad...

File: source/kraken/data_manager.h
Comment (xlqian) [reply source/kraken/data_manager.h:141]: I tried... like really hard :D . I thought flann was using the boost serializer. It turns out that it uses a home made serializer....

**Review by kinnou02 — APPROVED** [review]

**Pull request merged**
